Gems/ROS2/Code/Source/RobotImporter/FixURDF/FixURDF.cpp

```diff
@@ -0,0 +1,145 @@
+
+        return modifiedLinks;
+    }
+
+    //! Handles a case of multiple joints and the link sharing a common names which causes SDF error2 (but is fine in URDF)
+    //! Function will add a suffix "_dup" to the name of the joint if it is also the name of a link.
+    //! If there are name collisions in links, this will not be able to fix it, the SDF parser will throw an error.
+    //! @param urdf URDF to modify.
+    //! @returns a list of links that were modified
+    AZStd::vector<AZStd::string> RenameDuplicatedJoints(AZ::rapidxml::xml_node<>* urdf)
+    {
+        using namespace AZ::rapidxml;
+        AZStd::vector<AZStd::string> modifiedLinks;
+        AZStd::unordered_map<AZStd::string, unsigned int> linkAndJointsName;
+        for (xml_node<>* link = urdf->first_node("link"); link; link = link->next_sibling("link"))
+        {
+            auto* name = link->first_attribute("name");
+            if (name)
+            {
+                linkAndJointsName.insert(AZStd::make_pair(name->value(), 0));
+            }
+        }
+        for (xml_node<>* joint = urdf->first_node("joint"); joint; joint = joint->next_sibling("joint"))
+        {
+            auto* name = joint->first_attribute("name");
+            if (name)
+            {
+                if (linkAndJointsName.contains(name->value()))
+                {
+                    unsigned int& count = linkAndJointsName[name->value()];
+                    auto newName = AZStd::string::format("%s_dup%d", name->value(), count);
+                    name->value(urdf->document()->allocate_string(newName.c_str()));
+                    count++;
+                    modifiedLinks.push_back(AZStd::move(newName));
+                }
+                else
+                {
+                    linkAndJointsName.insert(AZStd::make_pair(name->value(), 0));
+                }
+            }
+        }
+        return modifiedLinks;
+    }
+
+    AZStd::pair<std::string, AZStd::vector<AZStd::string>> ModifyURDFInMemory(const std::string& data)
+    {
+        AZStd::vector<AZStd::string> modifiedElements;
+        using namespace AZ::rapidxml;
+        xml_document<> doc;
+        doc.parse<0>(const_cast<char*>(data.c_str()));
+        xml_node<>* urdf = doc.first_node("robot");
+        auto links = AddMissingInertiaToLinks(urdf);
+
+        if (links.size())
+        {
+            AZ_Warning("ROS2", false, "Added missing inertia to links: ");
+            for (auto& link : links)
+            {
+                modifiedElements.push_back(link);
+                AZ_Warning("ROS2", false, "  -> %s", link.c_str());
+            }
+        }
+
+        auto renames = RenameDuplicatedJoints(urdf);
+        if (renames.size())
+        {
+            AZ_Warning("ROS2", false, "Renamed duplicated links and joints: ");
+            for (auto& dup : renames)
+            {
+                modifiedElements.push_back(dup);
+                AZ_Warning("ROS2", false, "  -> %s", dup.c_str());
```

Comment:
I recommend not outputting any messages here, as the caller gets back all of the links that had missing inertia added and the renamed joints as part of the return value of this function.

The caller themselves can then log the messages, if they please.
This helps prevent console spam and allows the caller to choose to output or suppress the warning messages more easily for their use cases.